gnomad/utils/constraint.py

```diff
@@ -360,3 +412,290 @@ def collapse_strand(
+        )
+
+    # Filter to sites with coverage below `cov_cutoff` and larger than 0.
+    low_cov_ht = coverage_ht.filter(
+        (coverage_ht.exome_coverage < cov_cutoff) & (coverage_ht.exome_coverage > 0)
+    )
+
+    # Create a metric that represents the relative mutability of the exome calculated
+    # on high coverage sites and will be used as scaling factor when building the
+    # coverage model.
+    high_coverage_scale_factor = high_cov_ht.aggregate(
+        hl.agg.sum(high_cov_ht.observed_variants)
+        / hl.agg.sum(high_cov_ht.possible_variants * high_cov_ht.mu_snp)
+    )
+
+    # Generate a Table with all necessary annotations (x and y listed above)
+    # for the coverage model.
+    low_cov_group_ht = low_cov_ht.group_by(
+        log_coverage=hl.log10(low_cov_ht.exome_coverage)
+    ).aggregate(
+        low_coverage_oe=hl.agg.sum(low_cov_ht.observed_variants)
+        / (
+            high_coverage_scale_factor
+            * hl.agg.sum(low_cov_ht.possible_variants * low_cov_ht.mu_snp)
+        )
+    )
+
+    # Build the coverage model.
+    # TODO: consider weighting here as well
+    coverage_model_expr = build_coverage_model(
+        low_coverage_oe_expr=low_cov_group_ht.low_coverage_oe,
+        log_coverage_expr=low_cov_group_ht.log_coverage,
+    )
+    coverage_model = tuple(low_cov_group_ht.aggregate(coverage_model_expr).beta)
+    return coverage_model, plateau_models
+
+
+def build_plateau_models(
+    cpg_expr: hl.expr.BooleanExpression,
+    mu_snp_expr: hl.expr.Float64Expression,
+    observed_variants_expr: hl.expr.Int64Expression,
+    possible_variants_expr: hl.expr.Int64Expression,
+    pops_observed_variants_array_expr: List[hl.expr.ArrayExpression] = [],
+    weighted: bool = False,
+) -> Dict[str, Union[Dict[bool, hl.expr.ArrayExpression], hl.ArrayExpression]]:
+    """
+    Build plateau models to calibrate mutation rate to compute predicted proportion observed value.
+
+    The x and y of the plateau models:
+    - x: `mu_snp` - mutation rate
+    - y: proportion observed ('observed_variants' or 'observed_{pop}' / 'possible_variants')
+
+    :param cpg_expr: BooleanExpression noting whether a site is a CPG site.
+    :param mu_snp_expr: Float64Expression of the mutation rate.
+    :param observed_variants_expr: Int64Expression of the observed variant counts.
+    :param possible_variants_expr: Int64Expression of the possible variant counts.
+    :param pops_observed_variants_array_expr: Nested ArrayExpression with all observed
+        variant counts ArrayNumericExpressions for specified populations. e.g., `[[1,1,
+        1],[1,1,1]]`. Default is None.
+    :param weighted: Whether to generalize the model to weighted least squares using
+        'possible_variants'. Default is False.
+    :return: A dictionary of intercepts and slopes of plateau models for all set and
+        for populations. The keys for this dictionary are 'total' and 'pop'. The values
```

Comment:
I guess your purpose is to emphasize the dictionary won't return separate aggregators for each population. I rephrased it a bit but didn't use "group" because I used it in `count_variants_by_group` and `downsampling_counts_expr` when grouping tables. We can definitely use "group" if you think it won't confuse people!

> these two lines (with 527-529 above) aggregate the plateau models into a struct, convert them into a dictionary, and then reformats them again into a struct, which is a bit redundant. using an if/else statement here (while not as clean) would avoid these unnecessary reformats

Reply:
I'm a little confused; I think your comment is linked to another comment (not the one included below). This edit looks good, though maybe this wording is a bit clearer:
```
A dictionary of intercepts and slopes of plateau models for all sites and (optionally) per population.
```

Reply:
Sorry I linked the wrong comment :( This is the one I wanted to link: 

> maybe "group" here is better than "population"?